### ISSUE TITLE
Unable to process large modules

### ISSUE THREAD

**Issue by mbwhite** [nknapp/analyze-module-size#9]
Tyring to profile the 'composer-client' module, I've broken the stack.

```
angeError: Maximum call stack size exceeded
    at Function.from (native)
    at Package.totalStats (/home/matthew/.nvm/versions/node/v8.9.4/lib/node_modules/analyze-module-size/src/Package.js:54:35)
    at pkgs.map.pkg (/home/matthew/.nvm/versions/node/v8.9.4/lib/node_modules/analyze-module-size/src/index.js:39:27)
    at Array.map (<anonymous>)
    at toArchy (/home/matthew/.nvm/versions/node/v8.9.4/lib/node_modules/analyze-module-size/src/index.js:38:23)
    at pkgs.map.pkg (/home/matthew/.nvm/versions/node/v8.9.4/lib/node_modules/analyze-module-size/src/index.js:44:14)
    at Array.map (<anonymous>)
    at toArchy (/home/matthew/.nvm/versions/node/v8.9.4/lib/node_modules/analyze-module-size/src/index.js:38:23)
    at pkgs.map.pkg (/home/matthew/.nvm/versions/node/v8.9.4/lib/node_modules/analyze-module-size/src/index.js:44:14)
    at Array.map (<anonymous>)
```

Happy to help!

**Comment by nknapp** [comment]
Doea this module have that many dependencies? Or could this be an endless recursion?

**Comment by nknapp** [comment]
I don't even get that far. After cloning the project, installing the dependencies and  running the analyzer, I get the error message 

```
Could not find "/inquirer/ansi-styles" in ["/","/@angular/animations","/@angular/cli/anymatch",...
```
Weird, because the `node_modules/color-convert/package.json` contains a `_requiredBy: "/inquirer/ansi-styles"` but there is no `node_modules/inquirer/node_module/ansi-styles`-dependency in the tree. 

I tried removing the strict check and adding a cycle-detector and I found a cycle:
```
["/babel-core","/babel-register","/babel-register/babel-core"] => /babel-register
```
This would be the reason then. Is this really valid? `babel-core` uses `babel-register` uses `babel-core`?

**Comment by nknapp** [comment]
The `composer`-project uses `lerna` which seems to do some custom stuff to the dependency tree (like pruning dependencies). This seems to be the reason for my missing dependencies in the previous comment.

The analyzer uses the `_requiredBy`- and `_location`-properties in the package.json to build the dependency tree. I don't think the recreated tree will be correct with `lerna`. Also, I don't know, if I want to support this in the future, on my own.

The output may not be completely correct. But have add a a cycle-detector and a test for cycles to the `cycles-and-missing-deps`-branch. Maybe you can try that one and give some feedback before I publish it.

**Comment by mbwhite** [comment]
Yes - that branch did indeed solve the problem with the size of the modules. 
My local testing follows the lines of 
```
npm init -y
npm install --save composer-client  # which is the main entrypoint
```

I need to bypass the issue of the fs-events - but that's a separate issue.

**Comment by nknapp** [comment]
Fixed in 414ada75d0e7d9ee6022d9efc807cb0e8ba9410a (v1.4.0)
